package/MDAnalysis/core/topologyattrs.py

```diff
@@ -2038,8 +2074,412 @@ def total_charge(group, compound='group'):
+            quad_tensor = np.empty((n_compounds, 3, 3), dtype=np.float64)
+            for compound_mask, atom_mask in zip(compound_masks, atom_masks):
+                quad_tensor[compound_mask, :, :] = [
+                    __quadrupole(coords[mask] - ref[compound_mask][i],
+                                 chgs[mask][:, None])
+                    for i, mask in enumerate(atom_mask)
+                ]
+
+        return quad_tensor
+
+    transplants[GroupBase].append(('quadrupole_tensor', quadrupole_tensor))
+
+    def quadrupole_moment(group, **kwargs):
+        r"""Quadrupole moment of the group according to :cite:p:`Gray1984`.
+         
+        .. math::
+            Q = \sqrt{\frac{2}{3}{\hat{\mathsf{Q}}}:{\hat{\mathsf{Q}}}}
+
+        where the quadrupole moment is calculated from the tensor double 
+        contraction of the traceless quadropole tensor :math:`\hat{\mathsf{Q}}`
+
+        Computes the quadrupole moment of :class:`Atoms<Atom>` in the group.
+        Quadrupole per :class:`Residue`, :class:`Segment`, molecule, or
+        fragment can be obtained by setting the `compound` parameter
+        accordingly.
+
+        Note that when there is an unsymmetrical plane in the molecule or 
+        group, the magnitude of the quadrupole moment is dependant on the 
+        ``center`` chosen and cannot be translated.
+
+        Parameters
+        ----------
+        group : obj
+            AtomGroup
+        wrap : bool, optional
+            If ``True`` and `compound` is ``'group'``, move all atoms to the
+            primary unit cell before calculation.
+            If ``True`` and `compound` is not ``group``, the
+            centers of mass of each compound will be calculated without moving
+            any atoms to keep the compounds intact.
+        unwrap : bool, optional
+            If ``True``, compounds will be unwrapped before computing their
+            centers.
+        compound : {'group', 'segments', 'residues', 'molecules', \
+                    'fragments'}, optional
+            If ``'group'``, a single quadrupole value returns. Otherwise, an
+            array of each :class:`Segment`, :class:`Residue`, molecule, or
+            fragment will be returned as an array of position vectors, i.e.
+            a 1d array.
+            Note that, in any case, *only* the positions of
+            :class:`Atoms<Atom>` *belonging to the group* will be taken into
+            account.
+        center : {'mass', 'charge'}, optional
+            Choose whether the dipole vector is calculated at the center of 
+            "mass" or the center of "charge", default is "mass".
+
+        Returns
+        -------
+        numpy.ndarray
+            Quadrupole moment(s) of (compounds of) the group in :math:`eÅ^2`.
+            If `compound` was set to ``'group'``, the output will be a single
+            value. Otherwise, the output will be a 1d array of shape ``(n,)``
+            where ``n`` is the number of compounds.
+
```

Comment:
Let's link to to the other relevant method.
```suggestion

See Also
--------
quadrupole_tensor


```